src/contracts/service-contract/service-contract.spec.ts

```diff
@@ -500,78 +452,78 @@ describe('ServiceContract', function test() {
     expect(call.data).to.deep.eq(sampleCall);
     const answerId = await sc2.sendAnswer(
-      contract, accounts[2], sampleAnswer, callId, call.data.metadata.author,
+      contract, identity2, sampleAnswer, callId, call.data.metadata.author,
     );
 
     // retrieve answer with first account
-    const answer = await sc2.getAnswer(contract, accounts[0], callId, answerId);
+    const answer = await sc2.getAnswer(contract, identity0, callId, answerId);
     expect(answer.data).to.deep.eq(sampleAnswer);
   });
 
   it('can create answers and gets only basic answer information if unable to decrypt', async () => {
-    const contract = await sc0.create(accounts[0], businessCenterDomain, sampleService1);
+    const contract = await sc0.create(identity0, businessCenterDomain, sampleService1);
     await sc0.inviteToContract(
-      businessCenterDomain, contract.options.address, accounts[0], accounts[2],
+      businessCenterDomain, contract.options.address, identity0, identity2,
     );
-    const contentKey = await sharing.getKey(contract.options.address, accounts[0], '*', 0);
+    const contentKey = await sharing.getKey(contract.options.address, identity0, '*', 0);
     await sharing.addSharing(
-      contract.options.address, accounts[0], accounts[2], '*', 0, contentKey,
+      contract.options.address, identity0, identity2, '*', 0, contentKey,
     );
-    const callId = await sc0.sendCall(contract, accounts[0], sampleCall, [accounts[2]]);
+    const callId = await sc0.sendCall(contract, identity0, sampleCall, [identity2]);
 
     // retrieve call with other account, create answer
-    const call = await sc2.getCall(contract, accounts[2], callId);
+    const call = await sc2.getCall(contract, identity2, callId);
     expect(call.data).to.deep.eq(sampleCall);
     const answerId = await sc2.sendAnswer(
-      contract, accounts[2], sampleAnswer, callId, call.data.metadata.author,
+      contract, identity2, sampleAnswer, callId, call.data.metadata.author,
     );
 
     // retrieve answer with first account
-    const answer0 = await sc0.getAnswer(contract, accounts[0], callId, answerId);
+    const answer0 = await sc0.getAnswer(contract, identity0, callId, answerId);
     expect(answer0.data).to.deep.eq(sampleAnswer);
-    const answers0 = await sc0.getAnswers(contract, accounts[2], callId);
+    const answers0 = await sc0.getAnswers(contract, identity0, callId);
     expect(answers0[answerId].data).to.deep.eq(sampleAnswer);
 
     // retrieve answer with random account
-    const answer1 = await sc1.getAnswer(contract, accounts[2], callId, answerId);
-    expect(answer1.data).to.deep.eq(null);
-    const answers1 = await sc1.getAnswers(contract, accounts[2], callId);
-    expect(answers1[answerId].data).to.deep.eq(null);
+    const answer1 = await sc1.getAnswer(contract, identity1, callId, answerId);
+    expect(answer1.data).to.deep.eq(undefined);
+    const answers1 = await sc1.getAnswers(contract, identity1, callId);
```

Comment:
```suggestion
    const answers1 = await sc1.getAnswers(contract, identity2, callId);
```

Reply:
I am not quite sure what's supposed to happen here, because retrieving the answer with sc1 actually does work. Which in turn makes the test fail